Server/JokesOnYou.Web.Api/Controllers/TagController.cs

```diff
@@ -1,19 +1,65 @@
-﻿using JokesOnYou.Web.Api.Services.Interfaces;
+﻿using JokesOnYou.Web.Api.DTOs;
+using JokesOnYou.Web.Api.Services.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
 using System;
-using System.Collections.Generic;
-using System.Linq;
+using System.Security.Claims;
 using System.Threading.Tasks;
 
 namespace JokesOnYou.Web.Api.Controllers
 {
+    [Route("api/[controller]/[action]")]
+    [ApiController]
+    [Authorize]
     public class TagController : ControllerBase
     {
         private readonly ITagService _tagService;
+        private readonly ILogger _logger;
+        private readonly UserManager<ApplicationIdentity> _userManager;
 
-        public TagController(ITagService tagService)
+        public TagController(ITagService tagService, ILogger<TagController> logger, UserManager<ApplicationIdentity> userManager)
         {
             _tagService = tagService;
+            _logger = logger;
+            _userManager = userManager;
+        }
+
+        [HttpDelete]
+        public async Task<IActionResult> DeleteTag(int tagId)
+        {
+            var tag = await _tagService.GetTag(tagId);
+            string userId = _userManager.GetUserId(User);
+            if (tag.UserReplyDTO.Id == userId || User.IsInRole("Admin"))
+            {
+                await _tagService.DeleteTag(tagId);
+                return Ok($"{tagId}");
+            }
+            else
+            {
+                return Unauthorized();
```

Comment:
Why would it be unauthorized if the first expression in the if condition is not met?

Reply:
But if the User is not the owner nor an Admin s/he's not authorized to do the action ? is this not the intended result ?
We can add a msg to this as well right.